playbooks/06-deploy-architecture.yml

```diff
@@ -276,6 +276,11 @@
         -n openstack
         nova-cell0-conductor-0
         nova-manage cell_v2 discover_hosts --verbose
+        --strict
```

Comment:
From the doc:
> If --strict is specified, the command will only return 0 if an unmapped host was discovered and mapped successfully.

So this change means that the discover_hosts will be called until at least 1 host is discovered. This could help a bit with the race condition between starting the nova-compute service by the DataPlaneDeployment, letting the nova-compute start up and report the first status to the DB, and between the ci_framework running the discover_hosts command to discover those computes. 

However it will not be the final solution due to the following. If there are more than one compute in the deployment then discovering one of them is enough for this change to move forward. Which means maybe not all the compute will be discovered.

The final solution requires either of the two directions:
* a) do not let the DataPlaneDeployment finish until the nova-compute not just started but also successfully reported its first status to the DB. 
* b) retry discover_hosts until not just one but all the deployed compute nodes are discovered.